File: 251003/Kapinskiy/Publisher/src/main/java/by/kapinskiy/Publisher/models/Issue.java
package by.kapinskiy.Publisher.models;


import jakarta.persistence.*;
import lombok.Data;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "tbl_issue")
@Data
public class Issue {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne()
    @JoinColumn(name = "userId", referencedColumnName = "id")
    private User user;

    @Column(name = "title", unique = true)
    private String title;

    @Column(name = "content")
    private String content;

    @Column(name = "created")
    private Date created;

    @Column(name = "modified")
    private Date modified;

    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(name = "tbl_issue_tag", joinColumns = @JoinColumn(name = "issue_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id"))
    private List<Tag> tags = new ArrayList<>();

}

File: 251002/Urbanovich/Lab1.Core/Models/Issue.cs
namespace Lab1.Core.Models
{
    public class Issue
    {
        public ulong CreatorId { get; } = 0;
        public string Title { get; } = string.Empty;
        public string Content { get; } = string.Empty;
        public DateTime Created { get; } = DateTime.MinValue;
        public DateTime Modified { get; } = DateTime.MinValue;
        private Issue(ulong creatorId, string title, string content) => (CreatorId, Title, Content) = (creatorId, title, content);
        static public Issue Construct(ulong creatorId, string title, string content)
        {
            return new Issue(creatorId, title, content);
        }

    }
}
